### DISCUSSION TITLE
Fetch products based upon custom content

### DISCUSSION THREAD

**Discussion by Lulololu** [swellstores/community#388] [Help] [answered]
Hi,

We would like to fetch products related to a specific brand ID. Brand ID is retrieved from custom content.
We managed to get it with :

`swell.products.get({
      content.brand_id: "60f693c34eb0420008ba1161",
      expand: ["collection_detail", "brand_detail"],
    }),`
    
    
  It sounds really hacky, and obviously  TypeScript is full of errors.
  Do we have an alternative ?
  
  Thank you in advance

**Comment by logeshswell** [comment, marked as answer]
To fetch the list of products, use `list()` instead of `get()`
```ts
const response = swell.products.list({
  "content.brand_id": "60f693c34eb0420008ba1161",
  expand: ["collection_detail", "brand_detail"],
});
```

**Reply by Lulololu** [reply]
Thanks for your help.
